inference/conflict.go

```diff
@@ -87,20 +97,38 @@ func (n *nilFlow) String() string {
 	lastNonnilNodeIndex := len(n.nonnilPath) - 1
 	n.nonnilPath[lastNonnilNodeIndex].reason = fmt.Sprintf("%s (must be NONNIL)", n.nonnilPath[lastNonnilNodeIndex].reason)
 
-	flow := ""
-	for _, nodes := range [...][]node{n.nilPath, n.nonnilPath} {
-		for _, nodeObj := range nodes {
-			flow += nodeObj.String()
-		}
-	}
-	return flow
+	return pathString(n.nilPath) + pathString(n.nonnilPath)
 }
 
 func (c *conflict) String() string {
 	consumerPos := c.nilFlow.nonnilPath[len(c.nilFlow.nonnilPath)-1].position
 	producerPos := c.nilFlow.nilPath[0].position
+
+	// build string for similar conflicts (i.e., conflicts with the same nil path)
+	similarConflictsString := ""
+	if len(c.similarConflicts) > 0 {
+		similarPos := ""
+		for _, s := range c.similarConflicts {
+			similarPos += fmt.Sprintf("\"%s\", ", s.nilFlow.nonnilPath[len(s.nilFlow.nonnilPath)-1].position.String())
+		}
+		// remove trailing comma and space
+		similarPos = strings.TrimSuffix(similarPos, ", ")
```

Comment:
Does Go not have a string join function or similar? It feels kind of hacky to build the string with commas and then remove/replace stuff, maybe even adding an index to the for loop and then appending either `", "` for `i < len - 2`, " and " for `i == len - 2` and nothing for `i == len - 1` would be better?

Reply:
Actually there is. I was not aware of it. Updated the above code to now use `strings.Join()`